INSS.EIIR.Functions/Functions/ExtractJobTrigger.cs

```diff
@@ -0,0 +1,86 @@
+using INSS.EIIR.Interfaces.DataAccess;
+using INSS.EIIR.Interfaces.Messaging;
+using INSS.EIIR.Models.ExtractModels;
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.Azure.WebJobs;
+using Microsoft.Azure.WebJobs.Extensions.Http;
+using Microsoft.Azure.WebJobs.Extensions.OpenApi.Core.Attributes;
+using Microsoft.Azure.WebJobs.Extensions.OpenApi.Core.Enums;
+using Microsoft.Extensions.Logging;
+using Microsoft.OpenApi.Models;
+using System;
+using System.Net;
+using System.Threading.Tasks;
+
+namespace INSS.EIIR.Functions.Functions
+{
+    public class ExtractJobTrigger
+    {
+        private readonly ILogger<ExtractJobTrigger> _logger;
+        private readonly IServiceBusMessageSender _serviceBusMessageSender;
+        private readonly IExtractRepository _eiirRepository;
+
+        public ExtractJobTrigger(
+            ILogger<ExtractJobTrigger> log,
+            IServiceBusMessageSender serviceBusMessageSender,
+            IExtractRepository eiirRepository)
+        {
+            _logger = log;
+            _serviceBusMessageSender = serviceBusMessageSender;
+            _eiirRepository = eiirRepository;
+        }
+
+        [FunctionName("ExtractJobTrigger")]
+        [OpenApiOperation(operationId: "Run", tags: new[] { "name" })]
+        [OpenApiSecurity("function_key", SecuritySchemeType.ApiKey, Name = "code", In = OpenApiSecurityLocationType.Query)]
+        [OpenApiParameter(name: "name", In = ParameterLocation.Query, Required = true, Type = typeof(string), Description = "The **Name** parameter")]
+        [OpenApiResponseWithBody(statusCode: HttpStatusCode.OK, contentType: "text/plain", bodyType: typeof(string), Description = "The OK response")]
+        public async Task<IActionResult> Run([HttpTrigger(AuthorizationLevel.Function, "get", "post", Route = null)] HttpRequest req)
+        {
+            _logger.LogInformation($"ExtractJobTrigger triggered at: {DateTime.Now}");
+
+            var today = DateOnly.FromDateTime(DateTime.Now);
+
+            var extractJobQueue = Environment.GetEnvironmentVariable("servicebus:extractjobqueue");
+            if (string.IsNullOrEmpty(extractJobQueue))
+            {
+                _logger.LogError("ExtractjobTrigger missing configuration servicebus:extractjobqueue");
+                return new BadRequestObjectResult("ExtractjobTrigger missing configuration servicebus:extractjobqueue");
+            }
+
+            // 1. Query Extract jobs table and check if extract created for today
```

Comment:
separate into services ... comments always raise my alarm ... if you need to explain then it's doing too much ;)

Reply:
Comments were for my benefit so I didn't forget the implementation, they are not required i'll remove